vortex-scalar/src/primitive.rs

```diff
@@ -103,6 +108,46 @@ impl<'a> PrimitiveScalar<'a> {
             }?)),
         }
     }
+
+    pub fn checked_sub(&self, other: &PrimitiveScalar) -> VortexResult<Self> {
+        if self.ptype != other.ptype {
+            vortex_bail!("Failed to subtract {} and {}", self.ptype, other.ptype)
+        }
+        let result_pvalue: Option<PValue> = if !self.ptype.is_float() {
+            // checked_sub only implemented by integer type
+            match_each_integer_ptype!(self.ptype, |$T| {
+                let lhs = self.as_::<$T>()?;
+                let rhs = other.as_::<$T>()?;
+                match (lhs, rhs) {
+                    (Some(lv), Some(rv)) => {
+                        lv.checked_sub(rv).map(PValue::from)
+                    },
+                    _ => None
+                }
+            })
+        } else {
+            match_each_float_ptype!(self.ptype, |$T| {
```

Comment:
I would just do regular subtraction for floats.  There's not exactly overflow behavior

Reply:
I think it's ok now